server/database/seed.js

```diff
@@ -54,32 +68,40 @@ const seedAddress = () => {
     const countryCode = faker.address.countryCode();
     const latitude = constants.coordinates[i].lat;
     const longitude = constants.coordinates[i].lng;
-    db.query('INSERT INTO addresses (street_address, city, state, zip, country, country_code, latitude, longitude) VALUES (?, ?, ?, ?, ?, ?, ?, ?);', [streetAddress, city, state, zip, country, countryCode, latitude, longitude]);
+    db.query('INSERT INTO addresses (street_address, city, state, zip, country, country_code, latitude, longitude) VALUES (?, ?, ?, ?, ?, ?, ?, ?);', [streetAddress, city, state, zip, country, countryCode, latitude, longitude], callback);
   }
 };
-seedAddress();
 
 /* ========== DESCRIPTIONS TABLE========== */
-const seedDescriptions = () => {
+const seedDescriptions = (callback) => {
   for (let i = 0; i < 100; i += 1) {
     const editorialOne = (faker.lorem.sentences(4));
     const editorialTwo = (faker.lorem.paragraphs(4));
     const descriptionOne = (faker.lorem.sentence(3, 12));
     const descriptionTwo = (faker.lorem.paragraphs(2));
     const descriptionThree = (faker.lorem.paragraphs(4));
-    db.query('INSERT INTO descriptions (editorial_text_one, editorial_text_two, description_text_one, description_text_two, description_text_three) VALUES (?, ?, ?, ?, ?);', [editorialOne, editorialTwo, descriptionOne, descriptionTwo, descriptionThree]);
+    db.query('INSERT INTO descriptions (editorial_text_one, editorial_text_two, description_text_one, description_text_two, description_text_three) VALUES (?, ?, ?, ?, ?);', [editorialOne, editorialTwo, descriptionOne, descriptionTwo, descriptionThree], callback);
   }
 };
-seedDescriptions();
 
-const linkTableIDs = () => {
+const linkTableIDs = (callback) => {
   for (let i = 1; i < 101; i += 1) {
     const countUp = i;
     const countDown = 101 - i;
-    db.query('INSERT INTO `full_listing` (name_id, descriptions_id, rules_id, addresses_id) VALUES (?, ?, ?, ?)', [countUp, countDown, countUp, countDown]);
+    db.query('INSERT INTO `full_listing` (name_id, descriptions_id, rules_id, addresses_id) VALUES (?, ?, ?, ?)', [countUp, countDown, countUp, countDown], callback);
   }
 };
 
-linkTableIDs();
+const a = tracker('linkTableIDs', 100, () => db.end());
+const b = tracker('seedDescriptions', 100, linkTableIDs, a);
+const c = tracker('seedAddress', 100, seedDescriptions, b);
+const d = tracker('seedRules', 100, seedAddress, c);
+const e = tracker('seedHostels', 100, seedRules, d);
+
+seedHostels(e);
```

Comment:
I love how you used linked callbacks and a tracker to track how many records are being inserted.  Really nice work.

A QQ before I begin- Are you planning on using this script in your final project to generate your production data? Or was this more an exercise for yourself?

If you're using it in the final project, it's probably worth working on the readability. (You'd likely want to be able to show off this code to employers). Some suggestions:
* Add descriptive comments about each section, especially the tracker.
* Give the variables a-e on lines 95-99 somewhat more descriptive names.
* Rename the "callback" param in each function definition to something more descriptive like "nextTracker".
* Get rid of the commented out code.

If you're planning on not using this script in your final project, it's probably not worth putting any more thought into it. Nice work all around. 🥇

Reply:
These are excellent points. But I don't plan on using the script for SDC. I was just exercising and tried to come up with some fast ways of keeping track of what's going on with the async calls that had no callbacks in the first place. I may only consult how the original script is using fake data methods (data types, data lengths, etc...).